File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    long long binary(int x){
        int left=0,right=x/2;
        long long ans=0;
        while(left<=right){
            long long mid=left+(right-left)/2;
            long long sq=mid*mid;
            if(sq==x){
                return mid;
            }
            else if(sq<x){
                ans=mid;
                left=mid+1;
            }
            else{
                right=mid-1;
            }
        }
        return ans;
    }
    int mySqrt(int x) {
        if(x==0){
            return 0;
        }
        else if(x==1){
            return 1;
        }
        else{
        return(binary(x));
    }}
};

File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if (x == 0 || x == 1) {
            return x;
        }
        int l = 1, r = x / 2, ans = 0;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            long sq = (long) mid * mid; 
            if (sq == x) {
                return mid;
            } else if (sq < x) {
                ans = mid;
                l = mid + 1; 
            } else {
                r = mid - 1; 
            }
        }
        return ans;
    }
}

